src/ScottPlot5/ScottPlot5/AxisPanels/XAxisBase.cs

```diff
@@ -14,7 +14,7 @@ public float Measure()
         if (!IsVisible)
             return 0;
 
-        if (!Range.HasBeenSet)
+        if (!Range.HasBeenSet || TickGenerator.Ticks.Length == 0)
             return SizeWhenNoData;
```

Comment:
I like this, but I think instead of `SizeWhenNoData` (which is seemingly just hardcoded to 15) this should return `axisLabelHeight`. This is probably also necessary even when the range hasn't been set, so this is a problem that preexists your change.

I guess I'd also like a screenshot of how the plot looks when there's no ticks, to make sure it still looks right (or as right as it can look), even though a slightly broken axis is preferable to throwing.

Reply:
I may be misunderstanding the discussion, but the intent of returning 15 is to cover the case where there are no ticks and no axis labels. It's not aesthetic for the edge of the data area to extend right to the edge of the image, mostly because tick labels on other axes tend to get cut off if this is one.

Also FWIW, the hard-coded 15 pixels of padding for empty axes is user settable, so maybe we can call it soft coded 💁

Reply:
My point was that there could be no ticks, but still an axis label. So we need to measure the axis label if its present and return its height.

Reply:
Ah, now I understand - thanks for the clarification! I'll make that fix in main and tag this PR in it 👍

Reply:
This should do the trick! 🚀 
https://github.com/ScottPlot/ScottPlot/commit/8a12c5debde4b54ee0431f0f52d32539c0826ad9

Reply:
@swharden I think from the user's perspective it would be beneficial to disallow zooming to the levels when no ticks can be generated. Not sure if this functionality exists already.

Reply:
hi @drolevar, I suspect you can achieve that with "axis rules" https://scottplot.net/demo/#axis-rules